### PR TITLE
Fix order and naming of Difficulty Adjust mod settings

### LINKED ISSUE #7387: Order/Naming of Difficulty Adjust settings are different from beatmap details
**Describe the bug:** 
The order of the settings and the names of the settings are different between the beatmap details listed in song select and the settings panel for Difficulty Adjust. This can cause some confusion when looking between the settings and the beatmap details.

**Screenshots or videos showing encountered issue:** 
Details in song select:
image
Difficulty Adjust settings:
image

**osu!lazer version:** 
2019.1227.0

**Logs:** 
Not applicable.

### PR DESCRIPTION
**Pull request by protheory8**
Fixes issue #7387 by changing visible names of settings and by changing order of settings to [Circle Size, HP Drain, Accuracy, Approach Rate].

I used the solution of the user ICantDrive, because it seems good enough for now.

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs [hunk 1]
```diff
@@ -32,17 +52,21 @@ public class OsuModDifficultyAdjust : ModDifficultyAdjust
 
         protected override void TransferSettings(BeatmapDifficulty difficulty)
         {
-            base.TransferSettings(difficulty);
+            // base.TransferSettings(difficulty);
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs]: Commented code should either be uncommented, or deleted. At current I don't know what the reason for suppressing this is, and it's likely an incorrect thing to do too.

File: osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ public abstract class ModDifficultyAdjust : Mod, IApplicableToDifficulty
 
         public override Type[] IncompatibleMods => new[] { typeof(ModEasy), typeof(ModHardRock) };
 
-        [SettingSource("Drain Rate", "Override a beatmap's set HP.")]
+        [SettingSource("HP Drain", "Override a beatmap's set HP.")]
```
Comment (bdach) [inline osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs]: What is the reason behind these name changes?

File: osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs [hunk 3]
```diff
@@ -20,6 +20,26 @@ public class OsuModDifficultyAdjust : ModDifficultyAdjust
             Value = 5,
         };
 
+        [SettingSource("HP Drain", "Override a beatmap's set HP.")]
+        new public BindableNumber<float> DrainRate { get; } = new BindableFloat
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs]: These properties should not be duplicated.

**Review by bdach — CHANGES_REQUESTED** [review]
I think this solution is incorrect - as [this comment mentions](https://github.com/ppy/osu/issues/7387#issuecomment-569521110) a more robust one is needed. I'm envisioning an `Order` property on the `SettingSource` attribute that then would be [used in `SettingSourceExtensions.CreateSettingsControls`](https://github.com/ppy/osu/blob/f3ec98d85f5fe0a5a925ca2f72ddbf7d55b59e7f/osu.Game/Configuration/SettingSourceAttribute.cs) to appropriately order the settings controls.

File: osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs (hunk 2 above)
Comment (smoogipoo) [reply osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs]: To make them consistent with the beatmap detail area. But that place should be changed instead - the mods were already named correctly.

**Comment by ProTheory8** [comment]
Thank you, I will try to implement your idea or come up with another solution.

File: osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs (hunk 1 above)
Comment (ProTheory8) [reply osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs]: Got it.

**Comment by ProTheory8** [comment]
Should I make a new PR when I'm done or should I add commits to this one?

**Comment by peppy** [comment]
Please make a new PR. This one is completely incorrect in its direction.

**Pull request closed without merging**